tools/docker-network/tests/dockerframework.go

```diff
@@ -369,192 +412,192 @@ func (d *DockerTestFramework) StopIssueCandidacyPayload(nodes ...*Node) {
-
-	issuerResp, congestionResp := d.PrepareBlockIssuance(ctx, clt, issuer.Address)
+func (d *DockerTestFramework) IssueCandidacyPayloadFromAccount(wallet *mock.Wallet) iotago.BlockID {
+	block, err := wallet.CreateAndSubmitBasicBlock(context.TODO(), "", mock.WithPayload(&iotago.CandidacyAnnouncement{}))
+	require.NoError(d.Testing, err)
 
-	return d.SubmitPayload(ctx, &iotago.CandidacyAnnouncement{}, issuerId, congestionResp, issuerResp)
+	return block.ID()
 }
 
-// CreateTaggedDataBlock creates a block of a tagged data payload.
-func (d *DockerTestFramework) CreateTaggedDataBlock(issuerID iotago.AccountID, tag []byte) *iotago.Block {
-	issuer := d.wallet.Account(issuerID)
+// CreateTaggedDataBlock creates and submits a block of a tagged data payload.
+func (d *DockerTestFramework) CreateTaggedDataBlock(wallet *mock.Wallet, tag []byte) *iotago.Block {
 	ctx := context.TODO()
-	clt := d.wallet.DefaultClient()
-
-	issuerResp, congestionResp := d.PrepareBlockIssuance(ctx, clt, issuer.Address)
 
-	return d.CreateBlock(&iotago.TaggedData{
+	return lo.PanicOnErr(wallet.CreateBasicBlock(ctx, "", mock.WithPayload(&iotago.TaggedData{
 		Tag: tag,
-	}, issuerID, congestionResp, issuerResp)
+	}))).ProtocolBlock()
 }
 
-func (d *DockerTestFramework) CreateBasicOutputBlock(issuerAccountID iotago.AccountID) (*iotago.Block, *iotago.SignedTransaction, *mock.OutputData) {
-	clt := d.wallet.DefaultClient()
+func (d *DockerTestFramework) CreateBasicOutputBlock(wallet *mock.Wallet) (*iotago.Block, *iotago.SignedTransaction, *mock.OutputData) {
 	ctx := context.Background()
 
-	fundsOutputID := d.RequestFaucetFunds(ctx, iotago.AddressEd25519)
+	fundsOutputData := d.RequestFaucetFunds(ctx, wallet, iotago.AddressEd25519)
 
-	input := d.wallet.Output(fundsOutputID)
-	signedTx := d.wallet.CreateBasicOutputFromInput(input, issuerAccountID)
+	signedTx := wallet.CreateBasicOutputFromInput(fundsOutputData)
+	block := lo.PanicOnErr(d.defaultWallet.CreateBasicBlock(ctx, "", mock.WithPayload(signedTx))).ProtocolBlock()
 
-	issuerResp, congestionResp := d.PrepareBlockIssuance(ctx, clt, issuerAccountID.ToAddress().(*iotago.AccountAddress))
-	block := d.CreateBlock(signedTx, issuerAccountID, congestionResp, issuerResp)
-
-	return block, signedTx, input
+	return block, signedTx, fundsOutputData
 }
 
 // CreateDelegationBlockFromInput consumes the given basic output, then build a block of a transaction that includes a delegation output, in order to delegate the given validator.
-func (d *DockerTestFramework) CreateDelegationBlockFromInput(issuerID iotago.AccountID, accountAdddress *iotago.AccountAddress, inputID iotago.OutputID) (iotago.DelegationID, iotago.OutputID, *iotago.Block) {
-	issuer := d.wallet.Account(issuerID)
+func (d *DockerTestFramework) CreateDelegationBlockFromInput(issuerID iotago.AccountID, accountAdddress *iotago.AccountAddress, input *mock.OutputData) (iotago.DelegationID, iotago.OutputID, *iotago.Block) {
 	ctx := context.TODO()
```

Comment:
Create and pass context directly in line 457 if not used anywhere else.
Also maybe use everywhere `context.Background()` instead of `context.TODO`, but I believe we never decided and mix them a lot